lib/prometheus_exporter/middleware.rb

```diff
@@ -44,14 +44,25 @@ def call(env)
       controller = params["controller"]
     end
 
-    @client.send_json(
+    obj = {
       type: "web",
       timings: info,
       queue_time: queue_time,
       action: action,
       controller: controller,
       status: status
-    )
+    }
+    labels = custom_labels(env)
+    if labels
+      obj = obj.merge(custom_labels: labels)
+    end
+
+    @client.send_json(obj)
+  end
+
+  # allows subclasses to add custom labels based on env
```

Comment:
very important ... we are making an api change here so we must add something to the README about how to consume this.